File: src/main/java/io/playground/orderservice/infrastructure/kafka/model/EventEnvelope.java
package io.playground.orderservice.infrastructure.kafka.model;

import io.playground.orderservice.infrastructure.persistence.eventstream.OutboxEntity;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class EventEnvelope {
    private String eventId;
    private String eventType;
    private String payload;
    private String traceId;
    private String occurredAt;

    public static EventEnvelope from(OutboxEntity event) {
        return new EventEnvelope(
                event.getEventId(),
                event.getEventType().name(),
                event.getPayload(),
                event.getTraceId(),
                event.getOccurredAt().toString()
        );
    }
}

File: src/main/kotlin/io/playground/orderservice/infrastructure/kafka/model/EventEnvelope.kt
package io.playground.orderservice.infrastructure.kafka.model

import io.playground.orderservice.infrastructure.persistence.eventstream.OutboxEntity

data class EventEnvelope(
    val eventId: String,
    val eventType: String,
    val payload: String,
    val traceId: String,
    val occurredAt: String,
) {
    companion object {
        @JvmStatic
        fun from(event: OutboxEntity): EventEnvelope = EventEnvelope(
            event.eventId,
            event.eventType.name,
            event.payload,
            event.traceId,
            event.occurredAt.toString(),
        )
    }
}
